liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/BasicSupport.java

```diff
@@ -209,17 +221,39 @@ protected void init() throws MojoExecutionException, MojoFailureException {
             else { // default to install from runtime artifact
                 assemblyArtifact.setType("zip");
-                if(assemblyArtifact.getGroupId() == null) {
-                    log.debug("Defaulting runtimeArtifact group id to 'io.openliberty'");
-                    assemblyArtifact.setGroupId("io.openliberty");
+
+                // check for liberty.runtime.groupId property which overrides any groupId set in the assemblyArtifact
+                if (libertyRuntimeGroupId != null && !libertyRuntimeGroupId.isEmpty()) {
+                    if (assemblyArtifact.getGroupId() != null) {
+                        log.info("The runtimeArtifact groupId " + assemblyArtifact.getGroupId() + " is overwritten by the liberty.runtime.groupId value "+ libertyRuntimeGroupId +".");
+                    } else {
+                        log.info("The liberty.runtime.groupId property value "+ libertyRuntimeGroupId +" is used for the runtimeArtifact groupId.");
+                    }
+                    assemblyArtifact.setGroupId(libertyRuntimeGroupId);
+                } else {
+                    if(assemblyArtifact.getGroupId() == null) {
+                        log.debug("Defaulting runtimeArtifact group id to 'io.openliberty'");
+                        assemblyArtifact.setGroupId("io.openliberty");
+                    }
                 }
-                if(assemblyArtifact.getArtifactId() == null) {
-                    log.debug("Defaulting runtimeArtifact artifact id to 'openliberty-kernel'");
-                    assemblyArtifact.setArtifactId("openliberty-kernel");
+
+                // check for liberty.runtime.artifactId property which overrides any artifactId set in the assemblyArtifact
+                if (libertyRuntimeArtifactId != null && !libertyRuntimeArtifactId.isEmpty()) {
+                    if (assemblyArtifact.getArtifactId() != null) {
+                        log.info("The runtimeArtifact artifactId " + assemblyArtifact.getArtifactId() + " is overwritten by the liberty.runtime.artifactId value "+ libertyRuntimeArtifactId +".");
+                    } else {
+                        log.info("The liberty.runtime.artifactId property value "+ libertyRuntimeArtifactId +" is used for the runtimeArtifact artifactId.");
+                    }
+                    assemblyArtifact.setArtifactId(libertyRuntimeArtifactId);
+                } else {
+                    if(assemblyArtifact.getArtifactId() == null) {
+                        log.debug("Defaulting runtimeArtifact artifact id to 'openliberty-kernel'");
+                        assemblyArtifact.setArtifactId("openliberty-kernel");
+                    }
                 }
                 
                 // check for liberty.runtime.version property which overrides any version set in the assemblyArtifact
-                if (libertyRuntimeVersion != null) {
+                if (libertyRuntimeVersion != null && !libertyRuntimeVersion.isEmpty()) {
```

Comment:
I think it would be good to update the default version range below to start with `21.0.0.9` now.